File: .idea/app/src/main/java/com/chevstrap/rbx/Models/APIs/Roblox/GameCreator.java
package com.chevstrap.rbx.Models.APIs.Roblox;

import org.json.JSONObject;

public class GameCreator {
    private final long id;
    private final String name;
    private final String type;
    private final boolean isRNVAccount;
    private final boolean hasVerifiedBadge;

    public GameCreator(JSONObject obj) {
        this.id = obj.optLong("id");
        this.name = obj.optString("name");
        this.type = obj.optString("type");
        this.isRNVAccount = obj.optBoolean("isRNVAccount");
        this.hasVerifiedBadge = obj.optBoolean("hasVerifiedBadge");
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getType() {
        return type;
    }

    public boolean isRNVAccount() {
        return isRNVAccount;
    }

    public boolean hasVerifiedBadge() {
        return hasVerifiedBadge;
    }
}

File: app/src/main/java/com/chevstrap/rbx/models/apis/roblox/GameCreator.kt
package com.chevstrap.rbx.models.apis.roblox

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class GameCreator(
    @SerialName("id")
    val id: Long,

    @SerialName("name")
    val name: String? = null,

    @SerialName("type")
    val type: String? = null,

    @SerialName("isRNVAccount")
    val isRNVAccount: Boolean,

    @SerialName("hasVerifiedBadge")
    val hasVerifiedBadge: Boolean
)